lang/semgrep-grammars/src/semgrep-solidity/grammar.js

```diff
@@ -32,8 +32,21 @@ module.exports = grammar(base_grammar, {
           );
         },
 
-      // Metavariables. No need to patch the identifier rule because
-      // Solidity already accepts '$' as part of an identifier
+        // Metavariables. No need to patch the identifier rule because
+        // Solidity already accepts '$' as part of an identifier
+
+        // Metavariables for Solidity versions
+        _pragma_version_constraint: ($, previous) => {
```

Comment:
Did you try instead to override solidity_version and just allow an identifier there?

Reply:
Yes, I tried `choice(previous, $.identifier)` for solidity_version, but it somehow becomes `(solidity_version (identifier))` instead of just identifier. So, not sure that it will work